src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/memberdays/MemberDaysReport.java

```diff
@@ -0,0 +1,312 @@
+        return description;
+    }
+
+    private void addReportDetails(MemberDaysReportData reportData) {
+        List<MemberDaysReportDetail> interimReportDetails = new ArrayList<>();
+        for (var submitEvent : submitEvents) {
+
+            if (!isValidCase(submitEvent)) {
+                continue;
+            }
+
+            var caseData = submitEvent.getCaseData();
+            log.info("Adding case {} to Member Days report", caseData.getEthosCaseReference());
+
+            var hearings = getValidHearings(caseData);
+            extractValidHearingsFromCurrentCase(hearings, caseData, interimReportDetails);
+        }
+        var sortedReportDetails = interimReportDetails.stream()
+            .sorted((o1, o2) -> o1.comparedTo(o2)).collect(Collectors.toList());
+        reportData.getReportDetails().clear();
+        sortedReportDetails.forEach(d -> reportData.getReportDetails().add(d));
+    }
+
+    private void extractValidHearingsFromCurrentCase(List<HearingTypeItem> hearings,
+                                                     CaseData caseData,
+                                                     List<MemberDaysReportDetail> interimReportDetails) {
+        log.info("Adding hearings from the current case {} to the valid hearings list",
+            caseData.getEthosCaseReference());
+
+        for (var hearingItem : hearings) {
+            var dateListedTypeItems = filterHearingsWithHeardStatus(hearingItem);
+            var dateListedItems = filterValidHearingDates(dateListedTypeItems);
+
+            if (!CollectionUtils.isEmpty(dateListedItems)) {
+                for (var listedItem : dateListedItems) {
+                    var hearingDate = getLocalDate(listedItem.getValue().getListedDate());
+                    if (hearingDate != null) {
+                        var reportDetail = new MemberDaysReportDetail();
+                        reportDetail.setSortingHearingDate(hearingDate.toString());
+                        reportDetail.setHearingDate(UtilHelper.formatCurrentDate(hearingDate));
+                        reportDetail.setEmployeeMember(hearingItem.getValue().getHearingEEMember());
+                        reportDetail.setEmployerMember(hearingItem.getValue().getHearingERMember());
+                        reportDetail.setCaseReference(caseData.getEthosCaseReference());
+                        reportDetail.setHearingNumber(hearingItem.getValue().getHearingNumber());
+                        reportDetail.setHearingType(hearingItem.getValue().getHearingType());
+                        reportDetail.setHearingClerk(caseData.getClerkResponsible());
+                        reportDetail.setHearingDuration(getHearingDuration(listedItem));
+                        reportDetail.setParentHearingId(hearingItem.getId());
+                        interimReportDetails.add(reportDetail);
+                    }
+                }
+            }
+        }
+    }
+
+    private LocalDate getLocalDate(String listedDate) {
```

Comment:
Have you checked to see if there is already a helper method in ecm-common to do all this? You've got quite a lot of code to convert a string to a date - this is quite a common thing to do in ECM

Reply:
Or use SimpleDateFormatter which is already used in other classes, a similar implementation should be work where you can specify the exact pattern

Reply:
I am having look.